Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs

```diff
@@ -139,7 +139,12 @@ public ISiteBuilder CreateBuilder(ITracer tracer, ILogger logger, IDeploymentSet
         private ISiteBuilder ResolveNonAspProject(string repositoryRoot, string projectPath, IDeploymentSettingsManager perDeploymentSettings)
         {
             string sourceProjectPath = projectPath ?? repositoryRoot;
-            if (IsNodeSite(sourceProjectPath))
+            if (IsFunctionApp(sourceProjectPath))
```

Comment:
To make sure we don't hijack any existing Node apps that are not meant to be Functions, can you change the `IsFunctionApp` implementation to check if the `FUNCTIONS_EXTENSION_VERSION` environment variable is defined (value doesn't matter)? This is the best way to guarantee it's a function app, without needing to make guesses based on files. Note that we may allow pushing functions apps later that don't have a `host.json`.